mathics/core/symbols.py

```diff
@@ -379,6 +379,7 @@ def __new__(cls, name, sympy_dummy=None):
             self = super(Symbol, cls).__new__(cls)
             self.name = name
             self.sympy_dummy = sympy_dummy
+            self.value = value
```

Comment:
A type annotation for `value` is missing

Reply:
Still, I do not understand why `Symbol` requires this property.

Reply:
Yes, but you could also say the same for the other parameters too, like `name` in the unction definition,  and sympy_dummy (and even `cls`). 

While I think it good to start adding more type annotations, there are tons of them that are missing. This is one of those things that to do everything everywhere might be a bit much right now.

I am curious why this one in particular caught your attention?

Reply:
OK, I digged a little bit and I think that the reason why @rocky added this "value" parameter was to make that special symbols like `True` and `False` have it. However, should not be simpler and clearer to add the attribute just to these two special symbols?

Reply:
> I am curious why this one in particular caught your attention? 

It was commited in this PR - not a good reason 😅

Reply:
> OK, I dug a little bit and I think that the reason why @rocky added this "value" parameter was to make that special symbols like `True` and `False` have it. However, should not be simpler and clearer to add the attribute just to these two special symbols?

@mmatera if you have a way you think is cleaner and better - great. Just put in a separate PR for this.

Reply:
I have an idea. I am going to try it tomorrow.